### PR TITLE
Properly handle context cancelation in LookupResources processing

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: internal/services/shared/errors.go [hunk 1]
```diff
@@ -129,6 +129,13 @@ func RewriteError(ctx context.Context, err error) error {
 	case errors.Is(err, context.DeadlineExceeded):
 		return status.Errorf(codes.DeadlineExceeded, "%s", err)
 	case errors.Is(err, context.Canceled):
+		err := context.Cause(ctx)
```
Comment (vroldanbet) [inline internal/services/shared/errors.go:132]: TIL there is actually a cause to cancellation

File: internal/graph/reachableresources.go [hunk 2]
```diff
@@ -270,6 +270,12 @@ func (crr *CursoredReachableResources) chunkedRedispatch(
 				return nil, nil
 			}
 
+			// If the number of results returned was less than the limit specified, then this is
+			// the final iteration and no cursor should be returned for the next iteration.
+			if rsm.len() < int(queryLimit) {
+				lastTpl = nil
```
Comment (vroldanbet) [inline internal/graph/reachableresources.go:276]: was the cursor being returned via API? can we add test for this?

File: internal/graph/lookupresources.go [hunk 3]
```diff
@@ -55,14 +55,14 @@ func (cl *CursoredLookupResources) LookupResources(
 	reachableContext, cancelReachable := context.WithCancel(newContextForReachable)
 	defer cancelReachable()
 
-	limits, lCtx := newLimitTracker(lookupContext, req.OptionalLimit)
+	limits, _ := newLimitTracker(lookupContext, req.OptionalLimit)
```
Comment (vroldanbet) [inline internal/graph/lookupresources.go:58]: 👍🏻  reaching the limit should cancel RR but not LR. Im surprised we didn't have a test that would causes this to trigger.

File: internal/services/shared/errors.go (hunk 1 above)
Comment (josephschorr) [reply internal/services/shared/errors.go:132]: As of Go 1.20

File: internal/graph/reachableresources.go (hunk 2 above)
Comment (josephschorr) [reply internal/graph/reachableresources.go:276]: No; it just meant an extra iteration was occurring

**Comment by josephschorr** [comment]
Updated

**Review by vroldanbet — APPROVED** [review]

**Pull request merged**